plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java

```diff
@@ -243,6 +253,676 @@ public void testFlushPartitionCache()
         computeActual(format("DROP TABLE %s", fullyQualifiedTestTableName));
     }
 
+    @Test
+    public void testPartitionProjectionOnVarCharColumnWithEnumProjectionAndStorageLocationTemplate()
+    {
+        String tableName = "nation_" + randomTableSuffix();
+        String fullyQualifiedTestTableName = getTestTableName(tableName);
+
+        computeActual(getCreateTableStatement(
```

Comment:
I think that a demo test with one partition key would be welcome to understand the functionality

Reply:
Ok will add such test for simplies projection type

Reply:
Added `testPartitionProjectionOnVarCharColumnWithSingleEnumProjection`